### ISSUE TITLE
warnings not appearing in checkstyle if SCA_CPPCHECK_EXTRA_SUPPRESS is enabled

### ISSUE THREAD

**Issue by jordanmk** [priv-kweihmann/meta-sca#2004]
**Describe the bug**
After f9bb68d4f133d7fa01e7da982af055e1aee85977, if SCA_CPPCHECK_EXTRA_SUPPRESS is enabled, the errors/warnings that appear in cppcheck/raw are not present in cppcheck/checkstyle.

In local.conf I have:
SCA_CPPCHECK_EXTRA_SUPPRESS += "cppcheck.cppcheck.wrongPrintfScanfArgNum \\
				cppcheck.cppcheck.objectIndex \\
				cppcheck.cppcheck.unusedFunction \\
				cppcheck.cppcheck.varFuncNullUB \\
				cppcheck.cppcheck.duplicateCondition \\
				cppcheck.cppcheck.ConfigurationNotChecked \\
				"
In the cppcheck/raw output directory I can confirm that the defined warnings are suppressed. All other warnings/errors are still there. However, the cppcheck/checkstyle output directory ends up with empty* xml files.

*by empty I mean they only have:
\<?xml version="1.0" ?>
\<checkstyle version="4.3"/>

Going back to a4fa5c52b027a3037c4ed3510ea9d2d4fcd1a7c2, the EXTRA_SUPPRESS option doesn't work (as expected), but the checkstyle output directory has non-empty xml files.

I haven't checked what happens for other tools apart from cppcheck.

**Used layer**
- hash of meta-sca layer: f9bb68d4f133d7fa01e7da982af055e1aee85977

**Comment by jordanmk** [comment]
I have noticed that the space after the final argument makes a difference. If for example I have:
SCA_CPPCHECK_EXTRA_SUPPRESS += "cppcheck.cppcheck.wrongPrintfScanfArgNum \\
cppcheck.cppcheck.objectIndex \\
cppcheck.cppcheck.unusedFunction \\
cppcheck.cppcheck.varFuncNullUB \\
cppcheck.cppcheck.duplicateCondition \\
cppcheck.cppcheck.ConfigurationNotChecked"_
then the checkstyle output contains the other remaining warnings.

I am not 100%, but this might also be the case for SCA_SUPPRESS_LOCALS. Having an extra space after the last argument there might even cause build errors, but I haven't gone too deep there.

**Comment by priv-kweihmann** [comment]
Okay then it's potentially coming out of a totally different direction - will check.
I try reproducing it yesterday and had no luck, but you're comment might explain that

**Comment by priv-kweihmann** [comment]
I'm having a hard time trying to reproduce this.

- Does it work for you to add these suppression to the `suppress` file of recipes-sca-rules/sca-recipe-cppcheck-rules-native/files/suppress?
- Does it work when you add `SCA_CPPCHECK_EXTRA_SUPPRESS` to the recipe scope only?

None of what I tried so far could reproduce the things you've seen.
Do you have the chance to try it on master branch?

**Comment by jordanmk** [comment]
I can easily reproduce this, I always rm -rf the sca folder in build before any run, and make sure sca is triggered for the build I am making (I usually build one recipe, always toggling one print in its source, to make sure compiling and sca checks are triggered).

* Switched to master branch, this is still reproducible the same way as on the warrior branch
* Still reproducible if I put the SCA_CPPCHECK_EXTRA_SUPPRESS in foo.bb (and remove it from local.conf)
* Putting the suppresions to the suppress file works fine

So in conclusion, what works and doesn't work for me:
**Empty checkstyle xml files when:**
* SCA_CPPCHECK_EXTRA_SUPPRESS ends with empty space after last argument

**Correct checkstyle xml files when:**
* SCA_CPPCHECK_EXTRA_SUPPRESS doesn't have any empy spaces after last argument
* Suppressions are stored in recipes-sca-rules/sca-recipe-cppcheck-rules-native/files/suppress

**Comment by priv-kweihmann** [comment]
Thanks for the update - I'll keep on digging

**Comment by priv-kweihmann** [comment]
Just out of curiosity: if you would write 
```
SCA_CPPCHECK_EXTRA_SUPPRESS += "cppcheck.cppcheck.ConfigurationNotChecked "
```
into a single line, does it still not work?
I'm trying that all afternoon and for me this works like expected - just to rule out any strange non-printable characters somewhere in that string

**Comment by jordanmk** [comment]
So just to confirm what you've probably figured out:
SCA_CPPCHECK_EXTRA_SUPPRESS += "cppcheck.cppcheck.ConfigurationNotChecked " works fine
SCA_CPPCHECK_EXTRA_SUPPRESS += "cppcheck.cppcheck.ConfigurationNotChecked \" doesn't work fine.

I checked out the new changes, and now there are warnings when the escape character is the last one used:
_do_sca_cppcheck_report: bad escape (end of pattern) at position 0_

**Comment by priv-kweihmann** [comment]
Okay interesting - will try to find a way to make this work (although I thought that these line breaks should be removed by the bitbake parser)

**Comment by priv-kweihmann** [comment]
Yeah that's interesting `A = "a b c \"` makes it past the bitbake parser although to be consider invalid - that should be worth opening a bug on their side.
I'll filter those out as a work around
